infra/Pos/Communication/Relay/Class.hs

```diff
@@ -57,23 +58,26 @@ data MempoolParams m where
       ) => Proxy tag -> m [key] -> MempoolParams m
 
 data InvReqDataParams key contents m = InvReqDataParams
-    { contentsToKey :: contents -> m key
+    { invReqMsgType :: !MsgType
```

Comment:
This allows for a straightforward `ClassifyMsg` instance. Can we do block propagation/relaying using the `InvReqDataParams`/`DataParams`/`PropagationMsg` style?